pallets/jobs/src/rpc.rs

```diff
@@ -53,9 +52,103 @@ impl<T: Config> Pallet<T> {
 
 						jobs.push(info);
 					}
+				} else {
+					continue
 				}
 			}
 		}
-		Ok(vec![])
+
+		Some(jobs)
+	}
+
+	/// Queries a job by its key and ID.
+	///
+	/// # Arguments
+	///
+	/// * `role_type` - The role of the job.
+	/// * `job_id` - The ID of the job.
+	///
+	/// # Returns
+	///
+	/// An optional `RpcResponseJobsData` containing the account ID of the job.
+	pub fn query_job_by_id(
+		role_type: RoleType,
+		job_id: JobId,
+	) -> Option<RpcResponseJobsData<T::AccountId>> {
+		if let Some(job_info) = SubmittedJobs::<T>::get(role_type, job_id) {
+			if !job_info.job_type.is_phase_one() {
+				let result = KnownResults::<T>::get(
+					job_info.job_type.get_role_type(),
+					job_info.job_type.clone().get_phase_one_id().unwrap(),
+				)
+				.unwrap();
+
+				let info = RpcResponseJobsData::<T::AccountId> {
+					job_id,
+					job_type: job_info.job_type,
+					participants: result.participants(),
+					threshold: result.threshold(),
+					key: Some(result.result),
+				};
+
+				return Some(info)
+			} else {
+				let info = RpcResponseJobsData::<T::AccountId> {
+					job_id,
+					job_type: job_info.job_type,
+					participants: None,
+					threshold: None,
+					key: None,
+				};
+
+				return Some(info)
+			}
+		}
+
+		None
+	}
+
+	/// Queries the phase one result of a job by its key and ID.
+	///
+	/// # Arguments
+	///
+	/// * `role_type` - The role of the job.
+	/// * `job_id` - The ID of the job.
+	///
+	/// # Returns
+	///
+	/// An `Option` containing the phase one result of the job, wrapped in an
+	/// `RpcResponsePhaseOneResult`.
+	pub fn query_phase_one_by_id(
+		role_type: RoleType,
+		job_id: JobId,
+	) -> Option<RpcResponsePhaseOneResult<T::AccountId>> {
+		if let Some(job_info) = SubmittedJobs::<T>::get(role_type, job_id) {
+			if !job_info.job_type.is_phase_one() {
```

Comment:
What if this ID corresponds to a phase 1 job? Seems it will return none. Seems odd no?

Reply:
iiuc that is the goal of this api, `query_phase_one_by_id`, if its not phase one should return None, is that correct @shekohex ?

Reply:
Correct.